accepted/2024/improve-support-for-packages-signed-with-trusted-signing.md

```diff
@@ -0,0 +1,155 @@
+# Improve support for NuGet packages that are signed with a Trusted Signing certificate
+
+- Author Name: Dirk Lemstra [dlemstra](https://github.com/dlemstra)
+- GitHub Issue: https://github.com/NuGet/NuGetGallery/issues/10027
+
+## Summary
+
+This proposal is about improving the support for NuGet packages that are signed with a Trusted Signing certificate. This proposal
+consists of three steps to improve the experience for users that sign their packages with Trusted Signing. These three steps can
+be released separately but I can also understand that the NuGet team would want to launch this as a complete feature.
+
+## Motivation
+
+Earlier this year [Trusted Signing](https://learn.microsoft.com/en-us/azure/trusted-signing/) was launched by Microsoft and 
+recently support for this was added to [Sign CLI](https://github.com/dotnet/sign). Packages signed with this can be uploaded
+to the NuGet Gallery. But because the certificate is only valid for a short time a user will probably always need to update their
+certificate on their account page. I would like to propose a set of changes to improve the experience for users that sign their
+NuGet package with Trusted Signing.
+
+## Explanation
+
+### Functional explanation
+
+### Step 1: Register EKU of certificate that was added by Trusted Signing
+
+#### How would this be implemented?
+
+The first step would be to store the EKU that is inside the certificate that was used to sign the NuGet package with Trusted
+Signing. The certificate of Trusted Signing is only valid for thee days but it contains an EKU to that can be used to get the
+identity of the owner of the certificate. There are two EKU's in the certificate. One that tells us that it was signed with
+Trusted Signing (`1.3.6.1.4.1.311.97.1.0`) and another one that contains the Public Trust identity (e.g. 
+`1.3.6.1.4.1.311.97.990309390.766961637.194916062.941502583`). More details about this can be found here:
+https://learn.microsoft.com/en-us/azure/trusted-signing/concept-trusted-signing-cert-management. 
+
+I would like to propose to link EKU's to the user in a similar way as that is now done with certificates:
+
+![EKU](images/trusted-signing-eku.png)
```

Comment:
I assume that this screenshot is from Trusted Signing website, but I don't know for sure.  It does not look like a mockup for NuGet.org.  I think it's important to call out that this is a screenshot from the Trusted Signing website not exactly what you're proposing for NuGet.org.

Reply:
This is a mermaid diagram? Are you referring to another image?